### DISCUSSION TITLE
How to Update TeslaMate with Postgres in Portainer

### DISCUSSION THREAD

**Discussion by LukeSkywalker993** [teslamate-org/teslamate#4394] [Q&A]
Hello everyone,

I am currently running TeslaMate 1.31.1 with Postgres 13. I’d like to update, but since I’m using Portainer, the process isn’t straightforward.

I followed the guide provided in the [official TeslaMate documentation](https://docs.teslamate.org/docs/maintenance/upgrading_postgres/) for upgrading Postgres. I performed a backup and restore according to the instructions, which seem to work properly. However, unlike the guide, I’m using docker exec instead of docker compose, but I assume this difference is not critical.

After completing the restore process, which appears to run without any issues, I need to adjust the database settings in Portainer. This is where I encounter a problem, as Portainer refuses to accept the change.

Does anyone happen to have a script or a detailed guide specifically for upgrading Postgres in a Portainer environment?

Thanks in advance!

**Comment by YanJRL** [comment]
Hi,
Are you using portainer on synology ?
I followed the instructions from [here  ](https://docs.teslamate.org/docs/maintenance/upgrading_postgres/)and it works. Slight difference is that you use docker-compose rather than docker compose.
So my command were the following (yourdirectory is the installation directory of teslamate)
`` docker-compose exec -T database pg_dump -U teslamate teslamate>./teslamate.bck
  cp teslamate.bck /volume1/homes/yourdirectory/
docker-compose pull
docker-compose exec -T database pg_dump -U teslamate teslamate>./teslamate2.bck
cp teslamate2.bck /volume1/homes/yourdiretory/
docker-compose down
docker volume rm "$(basename "$PWD")_teslamate-db"
 vi docker-compose.yaml =>I used that to change the postgres version in the file, but you can use a graphic editor. search for postgres, and replace the number you have there by 17
docker-compose up -d database
docker-compose exec -T database psql -U teslamate -d teslamate < teslamate2.bck
docker-compose start teslamate
docker-compose pull
docker-compose up -d

**Reply by LukeSkywalker993** [reply]
Hello,
no, i using portainer on debian running in Hyper-V.

thanks for you help! i will try it next weekend!

**Comment by LukeSkywalker993** [comment]
Hello, thanks again for your response. I just wanted to test it, but it fails partly because I don't have "docker-compose," but rather "docker compose." However, that can still be translated into the new "docker compose."

The problem is that I installed Teslamate via Portainer. I don’t have a folder with a `docker-compose.yml` file.  

Here’s a screenshot of how it looks in Portainer.  
Overall, Portainer is quite practical, but when I want to replace an entire container, it seems not so much.
![grafik](https://github.com/user-attachments/assets/2c90de18-2dc5-4898-a303-3d7369b89681)

![grafik](https://github.com/user-attachments/assets/ef2e988d-e390-4840-b299-e4e19d2f0fde)

![grafik](https://github.com/user-attachments/assets/3a0ba91a-c663-4baf-a5bc-a440175e87fe)
